android-sdk/src/main/java/com/mobilecoin/lib/FogSearchKeyProvider.java

```diff
@@ -0,0 +1,58 @@
+package com.mobilecoin.lib;
+
+import com.google.protobuf.ByteString;
+import com.mobilecoin.lib.exceptions.KexRngException;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TreeMap;
+
+class FogSearchKeyProvider {
+
+    //Map FogSeed to the number of times its search keys have been scaled
+    Map<FogSeed, Integer> fogSeeds;
+
+    FogSearchKeyProvider(Collection<FogSeed> fogSeeds) {
+        this.fogSeeds = new TreeMap<FogSeed, Integer>();
+        fogSeeds.stream().forEach(seed -> this.fogSeeds.put(seed, 0));
```

Comment:
`this` makes it hard to read
using `this.name = name` in constructors makes sense to everyone, but anything more complex should be avoided